cli/azd/pkg/tools/azcli/webapp.go

```diff
@@ -51,7 +51,8 @@ func (cli *azCli) appService(
 
 func isLinuxWebApp(response *armappservice.WebAppsClientGetResponse) bool {
 	if response.Properties != nil && response.Properties.SiteConfig != nil &&
-		response.Properties.SiteConfig.LinuxFxVersion != nil {
+		response.Properties.SiteConfig.LinuxFxVersion != nil &&
+		*response.Properties.SiteConfig.LinuxFxVersion != "" {
```

Comment:
I'm wondering what `az` is doing currently. I looked at [azure-cli](https://github.com/search?q=repo%3AAzure%2Fazure-cli+linuxFxVersion&type=code&p=1) and didn't find similar logic that would use this property. I'm wondering if that Logic Apps Standard apps will not have either one of these set correctly, so this still isn't 100% reliable. Unfortunately, we're in a hard place where the server APIs have made it almost near impossible for the client to perform the correct functional behavior in all cases -- and some variation of a "best effort" attempt is needed.

The safest way possible is probably to probe the deployment status endpoint and fall back to zip deployment status in cases of some error (probably a 404). I'd recommend checking the exact logic in az and follow those steps since they've likely been "battle tested" for edge cases.

Once again, I'd like to emphasize that these APIs and how they work present many difficult problems that the client now needs to safeguard against, and always result in some degradation for the user in edge cases. On that front, I sympathize very much with our difficult position of holding the end of the stick.